dustcloud/server.py

```diff
@@ -31,528 +28,527 @@
-				print("magic ok")
-				process_data(socket,data)
+    """
+    This class works similar to the TCP handler class, except that
+    self.request consists of a pair of data and client socket, and since
+    there is no connection the client address must be given explicitly
+    when sending data back via sendto().
+    """
+    clienthello = ""
+    ddid = 0
+    ctx = ""
+    device_id = ""
+    dname = ""
+    sended = 0
+    commandcounter = 0
+    lastmessageid = 0
+    connmode = "udp"
+
+    def sendmydata(self, data):
+        self.request.sendto(data, self.client_address)
+
+    def handle(self):
+        data = self.request[0].strip()
+        socket = self.request[1]
+        print("{} via udp wrote:".format(self.client_address[0]))
+        # print("> RAW: %s" % binascii.hexlify(data))
+        if len(data) < 32:
+            print("len < 32")
+            return
+        else:
+            if data[0:2] == bytes.fromhex("2131"):  # Check for magic bytes
+                print("magic ok")
+                # TODO: process data
+                # process_data(socket, data)
+            else:
+                print("Unknown message: {}".format(data))
 
 
 class TCPSimpleServer(socketserver.ThreadingMixIn, socketserver.TCPServer):
-	# Ctrl-C will cleanly kill all spawned threads
-	daemon_threads = True
-	# much faster rebinding
-	allow_reuse_address = True
+    # Ctrl-C will cleanly kill all spawned threads
+    daemon_threads = True
+    # much faster rebinding
+    allow_reuse_address = True
+
+    def __init__(self, server_address, request_handler_class):
+        socketserver.TCPServer.__init__(self, server_address, request_handler_class)
 
-	def __init__(self, server_address, RequestHandlerClass):
-		socketserver.TCPServer.__init__(self, server_address, RequestHandlerClass)
 
 class UDPSimpleServer(socketserver.ThreadingMixIn, socketserver.UDPServer):
-	# Ctrl-C will cleanly kill all spawned threads
-	daemon_threads = True
-	# much faster rebinding
-	allow_reuse_address = True
+    # Ctrl-C will cleanly kill all spawned threads
+    daemon_threads = True
+    # much faster rebinding
+    allow_reuse_address = True
+
+    def __init__(self, server_address, request_handler_class):
+        socketserver.UDPServer.__init__(self, server_address, request_handler_class)
 
-	def __init__(self, server_address, RequestHandlerClass):
-		socketserver.UDPServer.__init__(self, server_address, RequestHandlerClass)
 
-		
 if __name__ == "__main__":
-	tcpserver = TCPSimpleServer(("0.0.0.0", 80), SingleTCPHandler)
-	udpserver = UDPSimpleServer(("0.0.0.0", 8054), MyUDPHandler) ## disable UDP interface
-	# terminate with Ctrl-C
-	try:
-		tcpserver.serve_forever()
-		udpserver.serve_forever()
-	except KeyboardInterrupt:
-		sys.exit(0)
+    tcpserver = TCPSimpleServer(("0.0.0.0", 80), SingleTCPHandler)
+    udpserver = UDPSimpleServer(("0.0.0.0", 8054), MyUDPHandler)  # disable UDP interface

```

Comment:
Haven't seen any traffic on the UDP server yet with my Gen1 vacuum. What do we expect is sent on that channel?

Reply:
@dgiese Looks like the vacuum talks udp on port 8053, was it intentional to create a UDP server on port 8054? Is there some other communication expected on port 8054?

Reply:
Looks like the vacuum uses either TCP or UDP, depending on which one first answers correctly. I guess we can then remove the UDP handler, or create a server and just ignore the messages.

Reply:
Hi,
great work! ;)
Technically UDP is somehow redundant. The devices (not only vaccum, but others too) try to connect to ot.io.mi.com on port 8053 first. If that does not work, they try to connect to ott.io.mi.com on port 80. When i first develop dustcloud i used the UDP interface, as it is simpler to use (without need to holding connections). But later developed on the TCP interface. I am not a good python developer, therefore i never succeeded to make the UDP interface work as wanted. Howeer technically port 8053 might be a good alternative for people, who run it publically and do not want/cant use port 80.

Reply:
Thanks for clarifying. I'll see if I can get the UDP communication up an running.